test/Kestrel.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -169,6 +172,66 @@ public Task BadRequestIfHostHeaderDoesNotMatchRequestTarget(string requestTarget
             Assert.Equal(LogLevel.Information, maxLogLevel);
         }
 
+        [Fact]
+        public async Task HostHeaderAcceptedWithPeriod()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host: .",
```

Comment:
I'm adding `{ "GET /pub/WWW/", "." }` to `HttpParsingData.HostHeaderData`, so this will get tested via `RequestTests.MatchesValidRequestTargetAndHostHeader`.